### DISCUSSION TITLE
Is there any way to list events?

### DISCUSSION THREAD

**Discussion by vCassius** [kube-rs/kube#1286] [Q&A]
Is there any way to list events? cuz i wanner to list pod list which pod status unready,and same times i wanner to match pods event,when event unhealth i will mark and clean them

**Comment by clux** [comment]
You can list and watch anything in k8s-openapi that implements `Resource`.
There happens to be examples for events a few places:
- https://github.com/kube-rs/kube/blob/main/examples/event_watcher.rs
- https://github.com/kube-rs/kube/blob/main/examples/node_watcher.rs#L27-L62
